extensions/community/CrazyGamesAdApi.json

```
@@ -0,0 +1,393 @@
+            "    const linkToShare = crazysdk.inviteLink({ roomId: roomId });",
+            "",
+            "    eventsFunctionContext.returnValue = linkToShare;",
+            "} catch(e) {",
+            "    console.log({e});",
+            "}"
+          ],
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": true
+        }
+      ],
+      "expressionType": {
+        "type": "string"
+      },
+      "parameters": [
+        {
+          "description": "RoomId",
+          "name": "RoomId",
+          "type": "string"
+        }
+      ],
+      "objectGroups": []
+    },
+    {
+      "description": "Check if CrazyGame SDK is loaded.",
+      "fullName": "Check if CrazyGame SDK is loaded",
+      "functionType": "Condition",
+      "name": "IsSDKLoaded",
+      "sentence": "Is CrazyGame SDK loaded?",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "eventsFunctionContext.returnValue = (CrazyGames != undefined);",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Display CrazyGames Happy Time.",
+      "fullName": "Display CrazyGames Happy Time",
+      "functionType": "Action",
+      "name": "DisplayHappyTime",
+      "sentence": "Display CrazyGames Happy Time",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "try {",
+            "    const crazysdk = window.CrazyGames.CrazySDK.getInstance();",
+            "    crazysdk.init();",
+            "",
+            "    // Player beats a boss, reaches a high score, etc.",
+            "    crazysdk.happytime();",
+            "} catch(e) {",
+            "    console.log({e});",
+            "}"
+          ],
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Check if user is using adblock.",
+      "fullName": "Check if user is using adblock",
+      "functionType": "Action",
+      "name": "ScanForAdBlockers",
+      "sentence": "Check if user is using adblock",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "try {",
+            "    function adblockDetection(event) {",
+            "        if (event.hasAdblock) {",
+            "            runtimeScene.getVariables().get(\"__CrazyGamesExtension\").getChild(\"adBlocker\").setBoolean(true);",
+            "        } else {",
+            "            runtimeScene.getVariables().get(\"__CrazyGamesExtension\").getChild(\"adBlocker\").setBoolean(false);",
+            "        }",
```

This can be simplified to:
```suggestion
            "            runtimeScene.getVariables().get(\"__CrazyGamesExtension\").getChild(\"adBlocker\").setBoolean(!!event.hasAdblock);",
```

---

updated